schema.c

```diff
@@ -1131,6 +1131,42 @@ sch_lookup (sch_instance * instance, const char *path)
     return lookup_node (instance, NULL, xmlDocGetRootElement (instance->doc), path);
 }
 
+void
+sch_lookup_node_get_ns_info (sch_instance * instance, const char *path, char **href, char **prefix)
+{
+    xmlNode *node = sch_lookup (instance, path);
+    if (node && node->ns)
+    {
+        if (node->ns->href)
+            *href = (char *) node->ns->href;
+        if (node->ns->prefix)
+            *prefix = (char *) node->ns->prefix;
+    }
+}
+
+char *
+sch_find_model_path_by_href_prefix (sch_instance * instance, char *href, char *prefix)
+{
+    xmlNode *xml = sch_child_first (instance);
+    char *path;
+    char *name;
+
+    while (xml && xml->type == XML_ELEMENT_NODE)
+    {
+        if (xml->ns &&
+            ((href && xml->ns->href && g_strcmp0 ((char *) xml->ns->href, href) == 0) ||
+             (!href && xml->ns->prefix && g_strcmp0 ((char *) xml->ns->prefix, prefix) == 0)))
+        {
```

Comment:
I am not sure what this function is trying to do. There is no such thing as a "model path" for a namespace

Reply:
This function finds the name of the root node for a model that matches the given namespace information

Reply:
I don't think there is such a thing as the "root node" for a model